src/benchmarklib/benchmark_runner.cpp

```diff
@@ -580,9 +573,7 @@ void BenchmarkRunner::_snapshot_segment_access_counters(const std::string& momen
 
   auto moment_or_timestamp = moment;
   if (moment_or_timestamp.empty()) {
-    const auto timestamp =
-        std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::system_clock::now() - _benchmark_start)
-            .count();
+    const auto timestamp = std::chrono::nanoseconds{std::chrono::steady_clock::now() - _benchmark_start}.count();
```

Comment:
Depends on what to cast to. If we casted to `std::chrono::seconds`, this would mean a loss. The thing is more that we want to explicitly have nanoseconds out of the difference, and the `.count()` depends on the type of the duration, as it is the number of ticks of the ratio defined by the duration template. As we want nanoseconds, we have to have a nanoseconds duration, and initialize a duration object here would lead to a compilation error if we lost precision or misused the chrono objects (which a `duration_cast` does not).